File: CS50X/Week 6/sentimental-mario-more/mario.py
def get_int(prompt):
    while True:
        try:
            value = int(input(prompt))
            if 1 <= value <= 8:
                return value
            else:
                print("Height must be a positive integer between 1 and 8.")
        except ValueError:
            print("Invalid input. Please enter a positive integer between 1 and 8.")


def printPyramid(height):
    for i in range(1, height + 1):
        # Print leading spaces for the left pyramid
        print(" " * (height - i), end="")
        # Print the left pyramid
        print("#" * i, end="")
        # Print the gap between the pyramids
        print("  ", end="")
        # Print the right pyramid
        print("#" * i)


def main():
    height = get_int("Height: ")
    printPyramid(height)


if __name__ == "__main__":
    main()

File: CS50X/Week 1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int bricks, int row);

int main(void)
{
    // Prompt the user for the pyramid's height
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n <= 0);

    // Print a pyramid of that height
    for (int i = 0; i < n; i++)
    {
        // Print row of bricks
        print_row(i, n);
    }
}

void print_row(int bricks, int row)
{
    // Print spaces
    for (int i = 0; i < row - bricks - 1; i++)
    {
        printf(" ");
    }

    // Print bricks
    for (int j = 0; j <= bricks; j++)
    {
        printf("#");
    }

    // Print spaces
    printf("  ");

    // Print bricks
    for (int k = 0; k <= bricks; k++)
    {
        printf("#");
    }

    printf("\n");
}
